python/lsst/dax/imgserv/cli.py

```diff
@@ -0,0 +1,162 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+'''
+This module implements the command line interface (CLI) to ImageServ.
+
+@author: Brian Van Klaveren, SLAC
+@author: Kenny Lo, SLAC
+
+'''
+
+import os
+import json
+
+import click
+
+import lsst.log as log
+import lsst.daf.persistence as dafPersist
+
+from .locateImage import (
+        image_open, W13DeepCoaddDb, W13RawDb, W13CalexpDb
+)
+
+from .image import Image
+from .dispatch import Dispatcher
+from .hashutil import Hasher
+
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+@click.command()
+@click.option("--config", help="Specify config file.")
+@click.option("--in_req", help="Specify query input.")
+@click.option("--out_dir", help="Specify the output directory.") 
+def exec_command(config_dir, in_req, out_dir):
+    """ Command Line Interface: Process query to return image file
+        in output directory.
+    """ 
+    cli = ImageServCLI(config_dir, in_req, out_dir)
+    cli.process_request()
+    
+class ImageServCLI(object):
+    """ Module to implement CLI for ImageServ.
+    
+    """
+    def __init__(self, config_dir, in_req, out_dir):
+        # load the configuration file
+        config = os.path.join(config_dir, "settings.json")
+        with open(config) as f:
+            self._config = json.load(f)
+            # strip /tests from dataroot if cur_dir is /tests
+            cwd = os.getcwd()
+            if "/tests" in cwd:
+                self._fix_dataroot_path()
+            f.close()
+         # configure the log file (log4cxx)
+        log.configure(os.path.join(config_dir, "log.properties"))
+        self._in_req = in_req
+        self._out_dir = out_dir
+        self._dispatcher = Dispatcher(os.path.join(ROOT, "config"))
+
+    def _fix_dataroot_path(self):
+        # this is needed to accommodate cur dir being different
+        # when running inside container image
+        img_ds = str(self._config["DAX_IMG_DS"])
+        img_dr = str(self._config["DAX_IMG_DR"])
+        self._config["DAX_IMG_DS"] = img_ds.replace("./tests", "./")
+        self._config["DAX_IMG_DR"] = img_dr.replace("./tests", "./")
+
+    def process_request(self):
+        errors, req = self._parse_req()
+        w13db = self._get_ds(req)
+        if w13db is not None:
+            img_getter = image_open(w13db, self._config)
+            # keep this specific instance of img_getter
+            Image.imagegetter = img_getter
+            result = self.dispatch(req)
+            name = req["name"]
+            fp_image = self._save_result(result, name) 
+            if self._check_result(req, fp_image):
+                print(name+": PASS")
+            else:
+                print(name+": FAIL")
+        else:
+            raise Exception("ImageServCLI", "w13db should NOT be None")
```

Comment:
It would be useful to have the image type returned in the exception. I'm assuming that this error message makes it back to the user.